### DISCUSSION TITLE
How to set all nodes in same displacement by BCs?

### DISCUSSION THREAD

**Discussion by LifengGan** [idaholab/moose#25961] [Q&A Modules: Solid mechanics] [answered]
Dear moose users, 

I want to let all nodes on one surface with same displacement in one paticular direction. But it is not fixed, just all node deformed in the same way in one direction.

How can I do that with BCs?

Best regards,
Lifeng

**Comment by GiudGiud** [comment]
Hello

You can set a DirichletBC on disp_x or disp_y for example.
Are you solving the solid mechanics equations? Are you solving for the total mesh deformation or are you simply modifying the location of nodes at a certain location?

Guillaume

**Reply by LifengGan** [reply]
I just want all the nodes on selected surface moving in the same way, more like a "weak" constrain to let nodes tie together

Cound it be done in the follwing way, by not define the value? I want those to be passive movement

```
[./right_v]
  type = DirichletBC
  variable = disp_x
  boundary = right
[../]
```

**Reply by GiudGiud** [reply]
it would not be done like that. DirichletBCs impose a value.
Do you want to impose the displacement value or just constrain the nodes to move together with the same amplitude / direction?

**Reply by LifengGan** [reply]
Yes, the latter, just constrain the nodes to move together with the same amplitude / direction

**Reply by GiudGiud** [reply, marked as answer]
You'll want to work in the Constraints system.

But we don't have this object right now. So you ll have to make a new one

**Reply by LifengGan** [reply]
Thanks a lot.

**Reply by LifengGan** [reply]
Hi, @GiudGiud 

Maybe it can be done this way?

```
[Constraints]
  [./z_top]
    type = EqualValueBoundaryConstraint
    variable = disp_z
    primary = '45' # node on top boundary
    secondary = 'top' # boundary
    penalty = 10e6
  [../]
[]
```

**Reply by GiudGiud** [reply]
So one node by one node? You can try but these constraints come at a cost (harder to solve)
It's also not very friendly if you have 100 nodes to set together, or if you go from one mesh to another mesh (all the ids change)

so I would develop a custom constraint instead

**Reply by LifengGan** [reply]
Here the idea is to use ` EqualValueBoundaryConstraint` to link the surface `top` and the node with id 45, which is also one node on top surface. And we those nodes in top surface can move in the same way.

If develop a custon constraint, which one i should take it as reference? and any suggestions on that?

**Reply by GiudGiud** [reply]
I havent worked much with Constraints unfortunately. 
I see @bwspenc @jbadger95 and @dschwen have contributed to that system, if they want to guide you
